src/nccl_ofi_net.c

```diff
@@ -931,46 +931,6 @@ int nccl_net_ofi_create_plugin(nccl_net_ofi_plugin_t **plugin_p)
 		NCCL_OFI_INFO(NCCL_INIT | NCCL_NET, "Using transport protocol %s (user set)",
 			      nccl_ofi_selected_protocol);
 	} else {
-		int num_accelerators = ofi_ndevices;
-
-#if HAVE_CUDA
-		if (nccl_net_ofi_cuDeviceGetCount(&num_accelerators) != CUDA_SUCCESS) {
-			NCCL_OFI_WARN("Error getting CUDA device count");
-			ret = -ENOTSUP;
-			goto exit;
-		}
-#endif
-
-		/* We try to use the RDMA protocol if all of the
```

Comment:
Deleting this block will essentially break P5 compiled without `--enable-platform-aws` (unless PROTOCOL is explicitly specified), which may or may not be a good thing. But we should consider at least warning when using more NICs than GPUs and not using RDMA protocol.

Reply:
I'm not sure what you mean by "will break"?  It's not going to make it any better or worse than it is today.

Reply:
Using the plugin on P5 without AWS optimizations works today, because this code block (that is being removed here) will pick the RDMA protocol.

Without this code block, the plugin without AWS optimizations will always default to the SENDRECV protocol, and our current failure mode with SENDRECV on P5 is an unhelpful segfault.

Reply:
"Works" is relative.  The right things will not happen because the PCIe speeds will be all fubared.  But more importantly, I don't know how to accomplish this block in the general world, so I'm totally willing to give that up.

Reply:
Ack. At some point I'd like to come up with a way to warn when running SENDRECV protocol in a configuration that will segfault, but that will take some more digging to figure out exactly what is happening there.